include/vcpkg/packagespec.h

```diff
@@ -145,16 +145,40 @@ namespace vcpkg
 
     struct Dependency
     {
+        struct Feature
+        {
+            std::string name;
+            PlatformExpression::Expr platform;
+            Feature(std::string name) : Feature(std::move(name), PlatformExpression::Expr::Empty()) { }
```

Comment:
I'm a bit nervous about this implicit conversion; too easy to accidentally chop the expression off.

Reply:
It is only relevant for the tests. It changes
```c++
{
     Dependency{"semver", {{"x"}}},
},
```
to
```c++
{
    Dependency{"semver", {Dependency::Feature{"x"}}},
},
```
I can change the 18 occurrences if you want.